src/Microsoft.AspNetCore.Sockets/Internal/AuthorizeHelper.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Security.Claims;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Internal;
+
+namespace Microsoft.AspNetCore.Sockets.Internal
+{
+    public static class AuthorizeHelper
+    {
+        public static async Task<bool> AuthorizeAsync(HttpContext context, AuthorizationPolicy policy)
```

Comment:
Given that we complete synchronously if `policy` is `null`, perhaps this is a place for `ValueTask<bool>`? C# 7 lets you drop it in place and continue using `async`. It's not a super high-performance code path, but it seems useful for the relatively common case of not having an auth policy.

@davidfowl thoughts on more `ValueTask`s? Premature optimization? Other unknown costs?